go/api/response_handlers.go

```diff
@@ -31,6 +31,76 @@ func handleStringOrNullResponse(response *C.struct_CommandResponse) string {
 	return handleStringResponse(response)
 }
 
+// handleIntegerOrNullResponse handles the response from a command that returns an integer or 0.
+func handleIntegerOrNullResponse(response *C.struct_CommandResponse) int64 {
+	if response == nil {
+		return 0
+	}
+
+	return int64(response.int_value)
+}
+
+// handleBooleanResponse converts a C struct_CommandResponse's bool_value to a Go bool.
+func handleBooleanResponse(response *C.struct_CommandResponse) bool {
+	if response == nil {
+		return false
+	}
+	return bool(response.bool_value)
+}
+
+// handleStringMapResponse processes a C.struct_CommandResponse to extract an array of Go strings.
+func handleStringMapResponse(response *C.struct_CommandResponse) map[string]string {
+	if response == nil { // Check if the response is nil
+		return nil
+	}
+
+	defer C.free_command_response(response)
+
+	// Preallocate slice for lengths.
+	lengths := unsafe.Slice(response.array_elements_len, response.array_value_len)
+	numElements := int(response.array_value_len)
+	if numElements == 0 || numElements%2 != 0 {
+		// Handle inconsistent array lengths (just in case)
```

Comment:
I would expect an error here if the array size does not divide by 2

Reply:
@eifrah-aws I've handled and improved it.